commands/utility/clearinfractions.js

```diff
@@ -0,0 +1,92 @@
+const Discord = require('discord.js');
+const dateFormat = require('dateformat');
+
+module.exports = {
+  name: 'clearinfractions',
+  description: 'Removes all infractions from a user',
+
+  execute(msg, con, args) {
+    const {status, err, userInfraction} = validInfraction(msg, args);
+    if (!status) {
+      return msg.reply(err);
+    }
+
+    clearInfractionSQL(msg, userInfraction, args, con);
+  },
+};
+
+function validInfraction(msg, args) {
+  const data = {
+    status: false,
+    err: null,
+    userInfraction: null,
+  };
+
+  if (!msg.member.roles.cache.some((role) => role.name === 'Admin')) {
+    data.err = msg.reply('You must be an Admin to use this command.');
+    return data;
+  }
+
+  data.userInfraction =
+    msg.mentions.members.first() || msg.guild.members.cache.get(args[0]);
+  if (!data.userInfraction) {
+    data.err = 'Please provide a user to remove infractions.';
+    return data;
+  }
+
+  data.status = true;
+  return data;
+}
+
+function clearInfractionSQL(msg, userInfraction, args, con) {
+  const now = new Date();
+  const date = dateFormat(now, 'yyyy-mm-dd HH:MM:ss');
+
+  const action = 'cc!clearinfractions ' + args.join(' ');
+
+  // Inserts row into database
+  const sql = `UPDATE infractions SET valid = ? WHERE user = ?;
+    INSERT INTO mod_log (timestamp, moderator, action, length_of_time, reason) VALUES
+    (?, ?, ?, NULL, NULL);`;
+
+  const values = [false, userInfraction.id, date, msg.author.id, action];
+  const escaped = con.format(sql, values);
+
+  con.query(escaped, function (err, result) {
+    if (err) {
+      console.log(err);
+    } else {
+      console.log('Infractions were removed from user');
```

Comment:
I think this should only be logged if infractions were actually removed (right now, it logs even if the user has no infractions).